docs/docs/how-to/performance/improving-site-performance.md

```diff
@@ -0,0 +1,203 @@
+- Javascript bundle size. 
+- Stylesheets and font files.
+- Images and other media. 
+- Resource requests & CDN caching configuration.
+
+Every site is different, and the recommendations will give some guidance as to where the highest effort to impact ratio is on your site. 
+
+### Address third-party-script impact
+
+Various types of calls made in your HTML, like calls to external font files, will block page load or page interactivity in different ways. In addition, third-party scripts can execute "eagerly", often delaying page load while they do so.
+
+#### First, lazy load all analytics plugins
+
+One of the lowest-hanging fruits is to set your scripts to load lazily rather than "eagerly" (the default). Any <script> tags being embedded manually can be set to <script async>.
+
+#### Second, inline scripts
+
+Rather than loading third-party scripts from external sources, you can inline scripts in your code to reduce the cost of a network call. There are a number of ways to do this. We recommend you [use onPreRenderHTML](https://www.gatsbyjs.org/docs/ssr-apis/#onPreRenderHTML) to add your script tag to the DOM; if possible put the scripts lower in your DOM so that they are parsed and evaluated later.
+
+Other options include:
+
+- If you want to defer execution even further, put it in [onClientEntry](https://www.gatsbyjs.org/docs/browser-apis/#onClientEntry) instead -- this will execute it after page load before the client renders.
+
+- if you are using [html.js](https://www.gatsbyjs.org/docs/custom-html/) instead, you can modify that fileto include your snippet.
+
+### Reduce your Javascript bundle cost
+
+Among all assets, Javascript can be uniquely costly to your performance. This is due to three main reasons.
+
+- First, like other assets, it needs to be loaded into your browser, 
+
+- Second, unlike most other assets, it's code. That means it needs to be parsed by your browser, which can "block" other work from happening. 
+
+- Third, with third-party npm modules, it's easy to accidentally add a lot of Javascript you don't actually need. 
+
+#### First, profile your bundle
+
+The first step to fix this is to figure out what's going on. Use gatsby-plugin-webpack-bundle-analyzer to profile your bundle. When you add this tool to your `gatsby-config.js`, you can analyze the bundle size on each page in your application.
+
+#### Second, remove unneeded third-party imports from commons.js
+
+Start by auditing your `commons.js` -- the bundle that is shared by all components.
+
+The first thing to do is inspect third-party package size. Anything over 50kb, and certainly 100kb, is worth examining whether it's needed. Some common culprits include: Moment.js ([see deprecation notice](https://momentjs.com/docs/#/-project-status/)), Lodash, Material UI (see [this description](https://www.freecodecamp.org/news/gatsby-perfect-lighthouse-score/#step3droppingmaterialuifortailwindcss)), but you'll want to inspect your individual libraries.
```

Comment:
These two links need more descriptive text (so that it's clear what each link goes to when the text is read out of context).

Reply:
thanks!